### ISSUE TITLE
muting master volume does not sync

### ISSUE THREAD

**Issue by AndyMutz2** [astralaster/volsync#2]
hi again!

first off, thank you SO MUCH for this tool, it makes using external USB DACs, which are not controllable by the Windows master volume, much more user friendly :)

in the current version 1.2 the volume sync does not seem to work, when the mute button is pressed. the master volume is muted, but the audio keeps playing at the same volume.

**Comment by astralaster** [comment]
Yes, that's exactly why I have created this tool. I also use a USB DAC which isn't controllable by Windows (Linux can do this btw). :) And older versions did work for you? I will look if I can reproduce the problem on my side.

**Comment by AndyMutz2** [comment]
just tried v1.0 and it has the same issue. I just propably did not try the mute button when I first tested v1.0, otherwise I would have reported it :)

**Comment by astralaster** [comment]
On my system (win10) it cuts the audio if I mute the master device. So it works even if I don't adjust the single streams.

**Comment by AndyMutz2** [comment]
I am on Win10, too.
that is weird, because when I press mute, every volume gets set to 0, so that works, but I do keep hearing the audio of whatever source is active.
when I mute the source specifically, then the audio gets cut.

![image](https://github.com/user-attachments/assets/db93c19c-f1c1-4b32-adde-c8cc43918b30)

EDIT: hmm, more weirdness happening.. I am pretty sure that muting the source did cut the audio, when I first tested this, but now even that does nothing and I still hear the audio.

**Comment by astralaster** [comment]
All sources going mute is normal. But on my system I cannot hear anything then. Sounds like a bug not only affecting volsync.

**Comment by AndyMutz2** [comment]
ok, I did some further testing:

muting just the active source works. I do not know why it stopped working, but after a reboot, muting the source cuts the audio.
no change when muting the master volume though, the audio keeps playing.

then I took a notebook with two different Windows installations (10 and 11) and tested those.
but the behaviour is the same as on my main PC.. muting the source works, muting the master volume does nothing.

just FYI, I am using the build in DAC in my Nubert nuPro A-300 speakers.

but what does work, is lowering the volume step by step to 0, then the audio gets cut.

would it be possible for you, to somehow implement this, so that when mute is pressed, the volume gets manually set to 0 and restored back to the former value when unmuting?

if this sounds too complicated, do not worry. my life does not depend on this ;)

**Comment by astralaster** [comment]
I'm also on nubert here, so sympathy wins. Could you try this version, if it solves your issue? This sets all session volumes to zero if the master volume gets muted and on unmute sets master volume again. I have also remove the console window, so it doesn't clutter the taskbar.
[VolSync.zip](https://github.com/user-attachments/files/17922252/VolSync.zip)

**Comment by AndyMutz2** [comment]
it works!
you are a legend :)
